src/geometry/manifold/instances/units_of_normed_algebra.lean

```diff
@@ -48,6 +49,115 @@ noncomputable theory
 
 open_locale manifold
 
+lemma open_embedding.to_local_homeomorph_left_inv
+ {α : Type*} {β : Type*} [topological_space α] [topological_space β]
+ (f : α → β) (h : open_embedding f) [nonempty α] {x : α} :
+ (open_embedding.to_local_homeomorph f h).symm (f x) = x :=
+begin
+  rw ←congr_fun (open_embedding.to_local_homeomorph_apply f h),
+  rw local_homeomorph.left_inv,
+  rw open_embedding.to_local_homeomorph_source,
+  apply set.mem_univ
+end
+
+lemma open_embedding.to_local_homeomorph_right_inv
+ {α : Type*} {β : Type*} [topological_space α] [topological_space β]
+ (f : α → β) (h : open_embedding f) [nonempty α] {x : β} (hx : x ∈ set.range f) :
+ f ((open_embedding.to_local_homeomorph f h).symm x) = x :=
+begin
+  rw ←congr_fun (open_embedding.to_local_homeomorph_apply f h),
+  rw local_homeomorph.right_inv,
+  rw open_embedding.to_local_homeomorph_target,
+  exact hx
+end
+
+lemma cont_mdiff_open_embedding
+  {𝕜 : Type*} [nontrivially_normed_field 𝕜]
+  {E : Type*} [normed_add_comm_group E] [normed_space 𝕜 E]
+  {H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+  {M : Type*} [topological_space M] [nonempty M]
+  {e : M → H} (h : open_embedding e) {n : with_top ℕ} :
+  @cont_mdiff _ _ _ _ _ _ _ I _ _ h.singleton_charted_space _ _ _ _ _ I _ _ _ n e :=
+begin
+  haveI := h.singleton_smooth_manifold_with_corners I,
+  rw cont_mdiff_iff,
+  split,
+  { apply h.continuous },
+  intros, -- show the function is actually the identity on the range of I ∘ e
```

Comment:
When using `intros` in the library, give variable names so that the code is robust against any future changes to the default-naming pattern.  So here
```suggestion
  intros x y, -- show the function is actually the identity on the range of I ∘ e
```
Same in a few other places.